experimental/modular_oak_functions/src/main.rs

```diff
@@ -0,0 +1,97 @@
+//
+// Copyright 2022 The Project Oak Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+//
+
+pub mod demux;
+pub mod io;
+pub mod log;
+pub mod lookup;
+pub mod policy;
+pub mod session;
+pub mod wasmi;
+
+use maplit::btreemap;
+use std::{collections::BTreeMap, sync::Arc};
+
+/// Manages the shared state for a service.
+pub trait Service {
```

Comment:
I feel like `SessionService` and `PolicyService` are a very specific types of `Service`. In particular, I guess, they'd never be passed to `WasmiService`, and will never be associated with an ABI function. So, I wonder if it is correct, conceptually, to have them as a `Service`.

Also, can we have an application that does not use `SessionService` or `PolicyService`?